### PR TITLE
MESH-1625: Move out `AssetName` from `SecurityToken`

### PR DESCRIPTION
**Pull request by Centril**
### modified API

- `SecurityToken.name` is moved into storage item `Asset.asset_names(ticker) => AssetName`.

### CODE REVIEW

File: pallets/asset/src/lib.rs [hunk 1]
```diff
@@ -1632,20 +1677,13 @@ impl<T: Config> Module<T> {
...
         }
-        ensure!(
-            funding_round
-                .as_ref()
-                .map_or(0u32, |name| name.len() as u32)
-                <= T::FundingRoundNameMaxLength::get(),
-            Error::<T>::FundingRoundNameMaxLengthExceeded
-        );
+        if let Some(fr) = &funding_round {
+            Self::ensure_founding_round_name_bounded(fr)?;
```
Comment (F-OBrien) [inline pallets/asset/src/lib.rs:1685]:
```suggestion
            Self::ensure_funding_round_name_bounded(fr)?;
```

File: pallets/asset/src/lib.rs [hunk 2]
```diff
@@ -1882,15 +1925,20 @@ impl<T: Config> Module<T> {
         ticker: Ticker,
         name: FundingRoundName,
     ) -> DispatchResult {
-        ensure!(
-            name.len() as u32 <= T::FundingRoundNameMaxLength::get(),
-            Error::<T>::FundingRoundNameMaxLengthExceeded
-        );
+        Self::ensure_founding_round_name_bounded(&name)?;
```
Comment (F-OBrien) [inline pallets/asset/src/lib.rs:1928]:
```suggestion
        Self::ensure_funding_round_name_bounded(&name)?;
```

File: pallets/asset/src/lib.rs [hunk 3]
```diff
@@ -1882,15 +1925,20 @@ impl<T: Config> Module<T> {
...
         let did = <ExternalAgents<T>>::ensure_perms(origin, ticker)?;
 
         FundingRound::insert(ticker, name.clone());
         Self::deposit_event(RawEvent::FundingRoundSet(did, ticker, name));
+        Ok(())
+    }
 
+    /// Ensure `name` is within the global limit for asset name lengths.
+    fn ensure_founding_round_name_bounded(name: &FundingRoundName) -> DispatchResult {
```
Comment (F-OBrien) [inline pallets/asset/src/lib.rs:1937]:
```suggestion
    fn ensure_funding_round_name_bounded(name: &FundingRoundName) -> DispatchResult {
```

**Review by F-OBrien — COMMENTED** [review]
I think `founding` was intended to be `funding` in `ensure_founding_round_name_bounded`

**Review by CJP10 — APPROVED** [review]

**Review by adamdossa — APPROVED** [review]

**Pull request merged**